hardware_interface/README.md

```diff
@@ -1,2 +1,132 @@
 robot agnostic hardware_interface package.
 This package will eventually be moved into its own repo.
+
+## Suggestions for `hardware_interface` implementation
+In the following section you can find some advices which will help you implement interface
+for your specific hardware.
+
+### Suggestions for having different update rate for each `hardware_interface` by counting loops
+Current implementation of [ros2_control main node](https://github.com/ros-controls/ros2_control/blob/master/controller_manager/src/ros2_control_node.cpp)
+has one update rate that controls the rate of the [`read()`](https://github.com/ros-controls/ros2_control/blob/fe462926416d527d1da163bc3eabd02ee1de9be9/hardware_interface/include/hardware_interface/system_interface.hpp#L169) and [`write()`](https://github.com/ros-controls/ros2_control/blob/fe462926416d527d1da163bc3eabd02ee1de9be9/hardware_interface/include/hardware_interface/system_interface.hpp#L178)
+calls in [`hardware_interface(s)`](https://github.com/ros-controls/ros2_control/blob/master/hardware_interface/include/hardware_interface/system_interface.hpp).
+In this section suggestion on how to run each interface implementation on its own update rate is introduced.
+
+1. Add parameters of main control loop update rate and desired update rate for your hardware interface.
+```
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
+
+  <xacro:macro name="system_interface" params="name main_loop_update_rate desired_hw_update_rate">
+
+    <ros2_control name="${name}" type="system">
+      <hardware>
+          <plugin>my_system_interface/MySystemHardware</plugin>
+          <param name="main_loop_update_rate">${main_loop_update_rate}</param>
```

Comment:
Can we avoid this somehow? Is this information only needed to `ResourceManager` to know when to trigger hardware?

Reply:
We could use the calculated `period` from the main loop for `ros2_control_node` and calculate the rate with `1/period`. What do you think?

Reply:
Yeah, we could, but it is probably difficult to send this for now.